### DISCUSSION TITLE
Loading Firebase configuration - Do I need also the firebase_options.dart?

### DISCUSSION THREAD

**Discussion by burekas7** [codemagic-ci-cd/codemagic-docs#2390] [Q&A]
### If you are looking for help, please confirm the following...

- [X] I confirm I have searched the [Docs](https://docs.codemagic.io), [Codemagic Sample Projects](https://github.com/codemagic-ci-cd/codemagic-sample-projects), and GitHub [Discussions](https://github.com/orgs/codemagic-ci-cd/discussions).
- [X] I confirm the software versions on my local computer match the ones in CI environment (e.g. Xcode version, Gradle version). See [here](https://docs.codemagic.io/troubleshooting/common-android-issues/#builds-work-locally-but-fail-on-codemagic)
- [X] I confirm I am able to do a fresh clone of the repository and build the project on my local computer.

### Which mobile framework are you using?

Flutter (Dart)

---

Hi,

I read the docs regarding Loading Firebase configuration:
https://docs.codemagic.io/knowledge-firebase/load-firebase-configuration/

It relates to those files:
- `google-services.json`
- `GoogleService-Info.plist`

But what about the `firebase_options.dart`?
It also contains api keys, and it used when initial Firebase:
```
    await Firebase.initializeApp(
      options: DefaultFirebaseOptions.currentPlatform,
    );
```
Do I need to add it also into environment variables?

---

Another question:
- Do I need to convert it to base64 by myself before pasting it, or just checking the "secure" checkbox and it will do it for me?
- Also I don't need to decode the value in the pre-build script, right? (Just as mentioned in docs?)

**Comment by burekas7** [comment]
Regarding using aws secret manager:
1) The script should be like that?
```
set -e # exit on first failed command

echo "$(aws secretsmanager get-secret-value \
  --secret-id ANDROID_FIREBASE_SECRET | \
  jq -r '.SecretString' | \
  jq -r '.ANDROID_FIREBASE_SECRET')" >> $CM_BUILD_DIR/android/app/google-services.json
  
echo "$(aws secretsmanager get-secret-value \
  --secret-id IOS_FIREBASE_SECRET | \
  jq -r '.SecretString' | \
  jq -r '.IOS_FIREBASE_SECRET')" >> $CM_BUILD_DIR/ios/Runner/GoogleService-Info.plist
```

2) If I have already script in the pre-build section with flutter commands, this secrets script should come before, right?

3) If I have a base64 value as a secret, should I need to use `SecretBinary` instead? Do I need to encode it?

**Reply by himesh-cm** [reply]
Hi @burekas7, 

After cloning the repository, make sure to decrypt it. This will allow you to use it in your workflow later. 
After fetching it from AWS, you can decode it base64 vars.

for example
```
echo "$(aws secretsmanager get-secret-value \
  --secret-id ANDROID_FIREBASE_SECRET | \
  jq -r '.SecretString' | \
  jq -r '.ANDROID_FIREBASE_SECRET' | base64 --decode)" >> $CM_BUILD_DIR/android/app/google-services.json
```

**Reply by burekas7** [reply]
@himesh-cm 
Thanks for your help.

So some updates regarding this try.

I got this error:
`You must specify a region. You can also configure your region by running "aws configure”.`

Even though I added the default region under my privat account global variables and secrets.
I followed this guide:
https://docs.codemagic.io/knowledge-others/import-variables-from-secret-manager/

Do I need to add the aws keys only in the app settings, and go to the Environment variables tab 
or can I do it also as I did in the global variable of my private account?

![image](https://github.com/codemagic-ci-cd/codemagic-docs/assets/13424257/9dd15237-76af-42c4-ad0d-f5f63312d969)

**Reply by himesh-cm** [reply]
@burekas7, Please share a build ID so I can further investigate the issue.

**Reply by burekas7** [reply]
@himesh-cm 

As I said:
It didn't work when I put the AWS variables in my Privat Account global variables:
build id: 65edb925fc7a5e71a4354b38
https://codemagic.io/app/636d13b239a6170009d6aedd/build/65edb925fc7a5e71a4354b38

But it does works when I remove it from the global iside into the specific App variables:
build id: 65eec9b07e8a3e4cb8572930
https://codemagic.io/app/636d13b239a6170009d6aedd/build/65eec9b07e8a3e4cb8572930

Thanks.

**Reply by burekas7** [reply]
@himesh-cm 
Any idea why?

**Comment by himesh-cm** [comment]
Hello @burekas7, 

I apologize for the delayed response, as the thread got missed. In regards to your initial query about the `firebase_options.dart` file, I would like to confirm that you can follow the same process as shown in the Codemagic documentation link you shared for `google-services.json` and `GoogleService-Info.plist`.

To ensure security, you can mark the environment as secure to encrypt the file. Encoding is not necessary.

**Comment by himesh-cm** [comment]
While I’m closing our discussion for now, if you have any other questions or suggestions, feel free to reply back or start a new discussion anytime. Have a nice day!

**Comment by ionicc** [comment]
Just in case this answer isn't clear to someone else, Passing of `options: DefaultFirebaseOptions.currentPlatform` is optional. In the case of it not being present, The SDK will automatically pick up `google-services.json` file in your android project or a `GoogleService-Info.plist` file in your iOS+ project and create one for you.

So, You can change this:
```dart
Firebase.initializeApp(options: DefaultFirebaseOptions.currentPlatform);
```
To
```dart
Firebase.initializeApp();
```
